cvmfs/magic_xattr.h

```diff
@@ -0,0 +1,251 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef CVMFS_MAGIC_XATTR_H_
+#define CVMFS_MAGIC_XATTR_H_
+
+#include <map>
+#include <string>
+
+#include "catalog_counters.h"
+#include "directory_entry.h"
+#include "shortstring.h"
+#include "util/string.h"
+
+class MountPoint;
+
+class BaseMagicXattr {
+  friend class MagicXattrManager;
+
+ public:
+  virtual bool PrepareValueFenced() { return false; }
+  virtual std::string GetValue() { return ""; }
+
+  virtual ~BaseMagicXattr() {}
+ protected:
+  BaseMagicXattr() {}
+
+  MountPoint *mount_point_;
+  PathString path_;
+  catalog::DirectoryEntry *dirent_;
+};
+
+class MagicXattrManager {
+ public:
+  explicit MagicXattrManager(MountPoint *mountpoint);
+  BaseMagicXattr *Get(const std::string &name, PathString path,
+                      catalog::DirectoryEntry *d);
+  std::string GetListString(catalog::DirectoryEntry *dirent);
+  void Register(const std::string &name, BaseMagicXattr *magic_xattr);
+
+ protected:
+  std::map<std::string, BaseMagicXattr *> base_xattrs_;
+  std::map<std::string, BaseMagicXattr *> withhash_xattrs_;
+  std::map<std::string, BaseMagicXattr *> regular_xattrs_;
+  std::map<std::string, BaseMagicXattr *> symlink_xattrs_;
+  std::map<std::string, BaseMagicXattr *> authz_xattrs_;
+  MountPoint *mount_point_;
+};
+
+class AuthZMagicXattr : public BaseMagicXattr {
+  virtual bool PrepareValueFenced();
```

Comment:
Doesn't this render the two virtual methods private, i.e. don't we need the `public` keyword?

Reply:
Good question! I haven't thought of that. I compared the magic xattr outputs between the old and the new version and it matched. So I guess it works as expected and override functions inherit the visibility as well.